src/flight_service/do_get.rs

```diff
@@ -289,38 +287,9 @@ mod tests {
         assert_eq!(stored_stage_keys.len(), 0);
     }
 
-    // Helper to create a mock physical plan proto
-    fn create_mock_physical_plan_proto(
-        partitions: usize,
-    ) -> datafusion_proto::protobuf::PhysicalPlanNode {
-        use datafusion_proto::protobuf::partitioning::PartitionMethod;
-        use datafusion_proto::protobuf::{
-            Partitioning, PhysicalPlanNode, RepartitionExecNode, Schema,
-        };
-
-        // Create a repartition node that will have the desired partition count
-        PhysicalPlanNode {
-            physical_plan_type: Some(
-                datafusion_proto::protobuf::physical_plan_node::PhysicalPlanType::Repartition(
-                    Box::new(RepartitionExecNode {
-                        input: Some(Box::new(PhysicalPlanNode {
-                            physical_plan_type: Some(
-                                datafusion_proto::protobuf::physical_plan_node::PhysicalPlanType::Empty(
-                                    datafusion_proto::protobuf::EmptyExecNode {
-                                        schema: Some(Schema {
-                                            columns: vec![],
-                                            metadata: std::collections::HashMap::new(),
-                                        })
-                                    }
-                                )
-                            ),
-                        })),
-                        partitioning: Some(Partitioning {
-                            partition_method: Some(PartitionMethod::RoundRobin(partitions as u64)),
-                        }),
-                    })
-                )
-            ),
-        }
+    // Helper to create a mock physical plan
+    fn create_mock_physical_plan(partitions: usize) -> Arc<dyn ExecutionPlan> {
+        let node = Arc::new(EmptyExec::new(SchemaRef::new(Schema::empty())));
+        Arc::new(RepartitionExec::try_new(node, Partitioning::RoundRobinBatch(partitions)).unwrap())
```

Comment:
This isn’t directly related to this PR, but might be something we consider updating or adding in a follow-up.

This `RoundRobinBatch` got me thinking 🤔. On one hand, it's a valid scenario we want our tests to cover. On the other, round-robin repartitioning isn’t ideal from a performance standpoint. How about we introduce a variety of partitioning schemes in our mock tests? That way, we can surface more realistic and appropriate cases while still including edge scenarios like this one.